frontend/src/libs/components/truck-filter/truck-filter.tsx

```diff
@@ -0,0 +1,66 @@
+import { type DeepPartial, type FieldValues } from 'react-hook-form';
+
+import { useAppForm, useCallback } from '~/libs/hooks/hooks.js';
+
+import { Dropdown } from '../dropdown/dropdown.js';
+import styles from './styles.module.scss';
+
+type Properties<T extends FieldValues> = {
+  defaultValues: DeepPartial<T>;
+  onSubmit: (payload: T) => void;
+};
+
+const TruckFilter = <T extends FieldValues = FieldValues>({
+  defaultValues,
+  onSubmit,
+}: Properties<T>): JSX.Element => {
+  const { control, handleSubmit } = useAppForm<T>({
+    defaultValues,
+  });
+
+  const handleFormSubmit = useCallback(
+    (event_: React.BaseSyntheticEvent): void => {
+      void handleSubmit(onSubmit)(event_);
+    },
+    [handleSubmit, onSubmit],
+  );
+
+  return (
+    <form className={styles.truckFilter} onSubmit={handleFormSubmit}>
```

Comment:
for a search we will need to listen form onChange + don't forget to debounce that call